### DISCUSSION TITLE
How to reference private/public cloud resources such as DNS infrastructure etc.

### DISCUSSION THREAD

**Discussion by petenorth** [open-gitops/project#84] [Q&A]
Taking the example of cert-manager.  We can choose to deploy to a cluster with Helm and then create and Issuer resource. Some of the values provided to the Issuer will depend on resources outside of the cluster .

For example 

https://cert-manager.io/docs/configuration/acme/dns01/azuredns/

When using GitOps the Issuer resource (and the cert-manager chart) will be present in the GitOps repo .

Should a GitOps technology provide a way to integrate with the cloud provider so that these parameters can be retrieved dynamically rather than being stored statically in the GitOps repo (directly from the cloud provider or from the infrastructure as code state)?

Or should a GitOps technology provide a standard way in which this information can be published to the GitOps repo ....

**Comment by lloydchang** [comment]
@petenorth GitOps and [external-dns](https://github.com/kubernetes-sigs/external-dns) operate independent control loops.  Are you asking to conflate the two ideas into one?  If yes, then why conflate?  Thanks.
